src/main/java/hudson/tasks/junit/JUnitResultArchiver.java

```diff
@@ -282,7 +284,12 @@ public static TestResultSummary parseAndSummarize(@Nonnull JUnitTask task, Pipel
             }
 
             if (!task.isSkipPublishingChecks()) {
-                new JUnitChecksPublisher(build, task.getChecksName(), result, summary).publishChecks(listener);
+                // If we haven't been provided with a checks name, and we have pipeline test details, set the checks name
+                // to be a ' / '-joined string of the enclosing blocks names. JUnitChecksPublisher will handle defaults if
+                // checksName ends up being empty or null
+                String checksName = task.getChecksName() != null || pipelineTestDetails == null ? task.getChecksName() :
+                      StringUtils.join(new ReverseListIterator(pipelineTestDetails.getEnclosingBlockNames()), " / ");
```

Comment:
Are the spaces around the slash conventional?

Reply:
I took my inspiration from https://github.com/jenkinsci/junit-plugin/blob/a39b176d80f8e984b9c8ec7ac22c4b2ac1d41b93/src/main/java/hudson/tasks/junit/CaseResult.java#L319

Reply:
Right…this is a display name for rendering in an HTML GUI, though, whereas a check name is in a somewhat different category because it is used as a key in the REST API, requested in branch protection settings, etc.

Reply:
I've been using slashes in my own custom checks without issue; nothing seems to mind.

>  used as a key in the REST API

As in in Githubs REST API?

Reply:
> I've been using slashes in my own custom checks without issue

Yes of course, I am just asking if there is a convention for using strings that look more like display names (`Tests / Branch / Stage`) or more like file paths or URIs (`tests/branch/stage`).

> As in in Githubs REST API?

Yes.

Reply:
> or more like file paths or URIs (tests/branch/stage).

I'm not sure I see where.

> file paths or URIs (tests/branch/stage).

ahh, to look more like this?

![image](https://user-images.githubusercontent.com/390569/99536396-35dc4180-29a2-11eb-8ba4-e5a33d0ec5aa.png)

Reply:
I prefer display name. something like `Tests / Unit`